### ISSUE TITLE
macosx_11_0_arm64 39 310 wheels: symbol not found in flat namespace '_H5get_libversion'

### ISSUE THREAD

**Issue by ngam** [Unidata/netcdf4-python#1161]
To report a non-security related issue, please provide:

* the version of the software with which you are encountering an issue
* environmental information (i.e. Operating System, compiler info, java version, python version, etc.)
* a description of the issue with the steps needed to reproduce it

If you have a general question about the software, please view our [Suggested Support Process](https://www.unidata.ucar.edu/support/#process).

---

In a clean python, installing netcdf4 from pypi doesn't work on osx-arm64 (m1 macs). To reproduce:
1. `mamba create -n netcdf4_wheel python=3.10`
2. `mamba activate netcdf4_wheel`
3. `pip install netcdf4`

  <details>
    <summary>Code output:</summary>

  ```
pip install netcdf4
Collecting netcdf4
  Using cached netCDF4-1.5.8-cp310-cp310-macosx_11_0_arm64.whl (464 kB)
Collecting cftime
  Using cached cftime-1.6.0-cp310-cp310-macosx_11_0_arm64.whl (195 kB)
Collecting numpy>=1.9
  Using cached numpy-1.22.3-cp310-cp310-macosx_11_0_arm64.whl (12.8 MB)
Installing collected packages: numpy, cftime, netcdf4
Successfully installed cftime-1.6.0 netcdf4-1.5.8 numpy-1.22.3
```

  </details>

4.  `python -c "import netCDF4"`

  <details>
    <summary>Code output:</summary>

```
python -c "import netCDF4"               
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/ngam/.Mambaforge-MacOSX-arm64/envs/netcdf4_wheel/lib/python3.10/site-packages/netCDF4/__init__.py", line 3, in <module>
    from ._netCDF4 import *
ImportError: dlopen(/Users/ngam/.Mambaforge-MacOSX-arm64/envs/netcdf4_wheel/lib/python3.10/site-packages/netCDF4/_netCDF4.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace '_H5get_libversion'

```

  </details>

The wheels from MacPython (https://github.com/MacPython/netcdf4-python-wheels/releases/tag/v1.5.8rel) work completely fine. Is it possible to retract the current wheels or update them asap?

Note: I tried this with python 3.9 and 3.10 from conda-forge as well as independent installation from python.org; same problem, but actually sometimes I got a different namespace that was problematic, `_nc_close` or something like that.

Thanks!

**Comment by ngam** [comment]
Oh... I just saw this: https://github.com/Unidata/netcdf4-python/pull/1159/commits/e91599d20347e01e81c9c49dce0b92fb7029b846#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R19. The pypi wheels prior to 1.6.0  were bad to begin with? That's odd --- why were they released then? 😅

**Comment by danielegrattarola** [comment]
Hey, sorry to resurrect a closed issue but is there any ETA for 1.6.0?
